### PR TITLE
#162170681 Add follow and unfollow functionality

### PR DESCRIPTION
**Pull request by Johnsonojo**
#### What does this PR do?
It has follow and unfollow functionality

#### Description of Task to be completed?
- create follow and unfollow actions
- create follow and unfollow reducers
- write tests for functionality

#### How should this be manually tested?
```bash
git clone git@github.com:andela/balder-ah-frontend.git

cd balder-ah-frontend

npm install

npm run start:dev
```

Visit `/profiles/:username` to follow or unfollow the author.

#### Any background context you want to provide?
A user should be able to follow other authors
A user should be able to unfollow other authors if they choose to

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170681](https://www.pivotaltracker.com/story/show/162170681)

#### Screenshots (If applicable)
foolllow 1
foolllow 2

#### Questions:
N/A

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -39,6 +40,8 @@ class App extends Component {
             <Route path="/articles" component={AllArticles} />
             <PrivateRoute path="/profile" component={ProfilePage} exact />
             <PrivateRoute path="/profile/articles" component={AllUserArticles} exact />
+            <Route path="/profile" exact component={ProfilePage} />
```
Comment (akhilome) [inline src/App.jsx]: Why this again. It has already been defined as a `PrivateRoute` on line 41

File: src/__tests__/actions/reactions/follow.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      expect(dispatch).toHaveBeenCalledWith({ type: GET_OTHER_FOLLOWERS, payload });
+    });
+
+    test('should not get all follower of a user', async () => {
+      const payload = { errors: { body: ['You have now unfollowed Johnson'] } };
+
+      await axiosMock.onPost().replyOnce(403, payload);
+      await getUserFollowers(username)(dispatch);
+      expect(dispatch).toHaveBeenCalledTimes(1);
+      //   expect(dispatch).toHaveBeenCalledWith({
```
Comment (akhilome) [inline src/__tests__/actions/reactions/follow.js]: Commented out code

File: src/components/profile/FollowAuthor.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+// import { withRouter } from 'react-router-dom';
```
Comment (akhilome) [inline src/components/profile/FollowAuthor.jsx]: Commented out chunk

**Review by akhilome — COMMENTED** [review]
Good job with this @Johnsonojo, I left some comments below 👇

File: src/App.jsx (hunk 1 above)
Comment (Johnsonojo) [reply src/App.jsx]: Thanks for spotting that. Will fix it

File: src/__tests__/actions/reactions/follow.js (hunk 2 above)
Comment (Johnsonojo) [reply src/__tests__/actions/reactions/follow.js]: Taken care of

File: src/components/profile/FollowAuthor.jsx (hunk 3 above)
Comment (Johnsonojo) [reply src/components/profile/FollowAuthor.jsx]: Resolved thanks

File: src/reducers/profile.js [hunk 4]
```diff
@@ -38,6 +40,15 @@ export default (state = initialState, action) => {
         ...state,
         error: action.payload.message,
       };
+    case type.GET_OTHER_PROFILE:
+      return {
+        ...state,
+        username: action.payload.userProfile.username,
```
Comment (oluwajuwon) [inline src/reducers/profile.js:46]: Nice work sof ar @Johnsonojo, wouldn't you rather destructure here?

**Review by oluwajuwon — COMMENTED** [review]
Nice job @Johnsonojo 👏🏿

I left a comment on your PR please have a look

**Review by frostyblok — APPROVED** [review]

**Review by akhilome — APPROVED** [review]

**Pull request merged**